### ISSUE TITLE
[UI] Submitted data: Dashboard donor table bulk select

### ISSUE THREAD

**Issue by kcullion** [icgc-argo/platform-ui#2321]
Note: this is a nice to have - we can prioritize it when we have time @b-f-chan 

## Goal
To allow a  program member to select a subset of donors and 
1. view the only the clinical data submitted for those donors
2. download only the submitted clinical data for those donors.

This is blocked by the submitted data section and filters being ready
